src/neynarAPI/NeynarAPIClient.ts

```diff
@@ -0,0 +1,679 @@
+    embeds?: CastEmbeds[]
+  ): Promise<PostCastResponseCast> {
+    const body: PostCastReqBody = {
+      signer_uuid: signerUuid,
+      text: text,
+      embeds: embeds,
+      parent: replyTo !== undefined ? replyTo.hash : undefined,
+    };
+    const response = await this.apis.v2Cast.postCast(body);
+    return response.data.cast;
+  }
+
+  /**
+   * Delete a cast
+   */
+  public async deleteCast(
+    signerUuid: string,
+    castOrCastHash: v2Cast | string
+  ): Promise<void> {
+    let castHash: string;
+    if (typeof castOrCastHash === "string") {
+      castHash = castOrCastHash;
+    } else {
+      castHash = castOrCastHash.hash;
+    }
+    const body: DeleteCastReqBody = {
+      signer_uuid: signerUuid,
+      target_hash: castHash,
+    };
+    await this.apis.v2Cast.deleteCast(body);
+  }
+
+  /**
+   * React to a cast
+   */
+  public async reactToCast(
+    signerUuid: string,
+    reaction: ReactionType,
+    castOrCastHash: v2Cast | string
+  ): Promise<OperationResponse> {
+    let castHash: string;
+    if (typeof castOrCastHash === "string") {
+      castHash = castOrCastHash;
+    } else {
+      castHash = castOrCastHash.hash;
+    }
+    const body: ReactionReqBody = {
+      signer_uuid: signerUuid,
+      reaction_type: reaction,
+      target: castHash,
+    };
+    const response = await this.apis.reaction.postReaction(body);
+    return response.data;
+  }
+
+  /**
+   * Remove a reaction to a cast
+   */
+  public async removeReactionToCast(
+    signerUuid: string,
+    reaction: ReactionType,
+    castOrCastHash: v2Cast | string
+  ): Promise<OperationResponse> {
+    let castHash: string;
+    if (typeof castOrCastHash === "string") {
+      castHash = castOrCastHash;
+    } else {
+      castHash = castOrCastHash.hash;
+    }
+    const body: ReactionReqBody = {
+      signer_uuid: signerUuid,
+      reaction_type: reaction,
+      target: castHash,
+    };
+    const response = await this.apis.reaction.deleteReaction(body);
+    return response.data;
+  }
+
+  /**
+   * Fetches casts in a given thread.
+   * Note that the parent provided by the caller is included in the response.
+   */
+  public async fetchCastsInThread(
+    threadParent: v1Cast | { hash: string }
+  ): Promise<v1Cast[] | undefined> {
+    const response = await this.apis.v1Cast.allCastsInThread(threadParent.hash);
+    return response.data.result.casts;
+  }
+
+  /**
+   * Fetch the latest cast for the user, if there is one
+   */
+  public async fetchLatestCastForUser(user: {
+    fid: number;
+  }): Promise<v1Cast | undefined> {
+    // eslint-disable-next-line no-unreachable-loop
+    for await (const cast of this.fetchCastsForUser(user, {
+      pageSize: 5,
+    })) {
+      return cast;
+    }
+    return undefined;
+  }
+
+  /**
+   * Gets all casts (including replies and recasts) created by the specified user.
+   *
+   * @Note: Deleted cast filtering is applied server-side while recast filtering is applied
+   * client-side.
```

Comment:
This note is referring to the parameters `includeDeletedCasts = false, includeRecasts = false` which seem to have been removed in the migration

Reply:
removed